### PR TITLE
feat(query): show columns support display view engine

### LINKED ISSUE #11136: Feature: hope  to support view use "show full COLUMNS from view_xx form xx"
use view can not show COLUMNS .
There is no problem with the display in mysql view
image
There is no problem with the display in table
image
image

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Because of #10632, now show columns can not query the view engine field. 

In this pr, add ShowColumnInterpreter to support this.

We will generate a data block and if `show columns` have a limit, will filter the data block.

Closes #11136

### CODE REVIEW

File: src/query/service/src/interpreters/interpreter_show_columns.rs [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+impl ShowColumnsInterpreter {
+    pub fn try_create(ctx: Arc<QueryContext>, plan: ShowColumnsPlan) -> Result<Self> {
+        Ok(ShowColumnsInterpreter { ctx, plan })
+    }
+}
+
+#[async_trait::async_trait]
+impl Interpreter for ShowColumnsInterpreter {
+    fn name(&self) -> &str {
+        "ShowCreateTableInterpreter"
```
Comment (b41sh) [inline src/query/service/src/interpreters/interpreter_show_columns.rs:67]: ShowColumnsInterpreter

**Review by youngsofun — APPROVED** [review]

File: src/query/service/src/interpreters/interpreter_show_columns.rs [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+                if let Some(expr) = field.default_expr() {
+                    default_expr = expr.to_string();
+                }
+                exprs.push(default_expr.into_bytes());
+                if field.is_nullable() {
+                    is_nullables.push("YES".to_string().into_bytes());
+                } else {
+                    is_nullables.push("NO".to_string().into_bytes());
+                }
+                extras.push("".to_string().into_bytes());
```
Comment (b41sh) [inline src/query/service/src/interpreters/interpreter_show_columns.rs:128]: this column seems always empty, how about use `BlockEntry` with a `Scalar` value. like:
```rust
BlockEntry {
    data_type: DataType::String,
    value: Value::Scalar(Scalar::String(vec![])),
}
```

File: src/query/service/src/interpreters/interpreter_show_columns.rs [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+                StringType::from_data(keys),
+                StringType::from_data(exprs),
+                StringType::from_data(extras),
+                StringType::from_data(privileges),
+                StringType::from_data(comments),
+            ])
+        } else {
+            // | Field | Type | Null | Key  | Default | Extra |
+            DataBlock::new_from_columns(vec![
+                StringType::from_data(names),
```
Comment (b41sh) [inline src/query/service/src/interpreters/interpreter_show_columns.rs:154]: It seems to have some repetitive code that could be simplified.

Comment (TCeason) [reply src/query/service/src/interpreters/interpreter_show_columns.rs:154]: I think we can not do. Need ensure that the sequence is the same as the schema

File: src/query/service/src/interpreters/interpreter_show_columns.rs (hunk 2 above)
Comment (TCeason) [reply src/query/service/src/interpreters/interpreter_show_columns.rs:128]: If do this, can not ensure that the sequence is the same as the schema?

**Comment by sundy-li** [comment]
`system.columns`  will not display view engine, but `show columns` will.

**Pull request closed without merging**